src/database-layer/db.ts

```diff
@@ -184,8 +187,16 @@ export abstract class Tx {
 	public executeSql(sql: Sql, bindings: Bindings = [], ...args: any[]): Promise<Result> {
 		this.incrementPending()
 
+		let t0 = Date.now();
 		return this._executeSql(sql, bindings, ...args)
-			.finally(() => this.decrementPending())
+			.finally(() => {
+				this.decrementPending()
+				const queryTime = (Date.now() - t0) / 1000;
+				metricsEmitter.emit('db_query_time', { 
+					queryTime: queryTime,
+					queryType: sql.substr(0, sql.indexOf(' '))
```

Comment:
It's possible to run multiple statements in one `executeSql` call by separating the different queries with `;`, it's also possible that the first line is a comment like `-- Some comment`, and also it could be that the user just has leading space, eg ` SELECT`, so this query type detection is going to be shaky due to that.  I'm not sure if there's a better way to do it here but it definitely comes with a bunch of caveats as it stands.

- this doesn't block the PR, it would just be nice to improve on if possible

Reply:
it's a pretty rough approximation, agreed. A poor approximation though it is, I've left it and added two comments tagged metrics-TODO: which point to two avenues of future improvement I've identified.